File: structy/02-linked-list/printLinkedList.js
export default function printLinkedList(head) {
    if (head === null) return;
    console.log(head.val);
    printLinkedList(head.next)
}

// export default function printLinkedList(head) {
//     let currentNode = head;
//     while (currentNode.next !== null) {
//         console.log(currentNode.val);
//         currentNode = currentNode.next;
//     }
// }

File: structy/02-linked-list/lib/print_linked_list.py
def print_list(head):
    result = []
    current_node = head
    while current_node is not None:
        result.append(str(current_node.val) + ' -> ')
        current_node = current_node.next

    joined_result = ''.join(result) + 'None'
    print(joined_result)